ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/LedgerApiServer.scala

```diff
@@ -148,6 +149,37 @@ final class LedgerApiServer(
         server
       })(server => Future(server.shutdown().awaitTermination()))
     }
+
+    // This exposes the existing services under com.daml also under com.digitalasset.
+    // This is necessary to allow applications built with an earlier version of the SDK
+    // to still work.
+    // The "proxy" services will not show up on the reflection service, because of the way it
+    // processes service definitions via protobuf file descriptors.
+    private def addService(serverBuilder: NettyServerBuilder, service: BindableService): Unit = {
+      // First add the original com.daml service
+      serverBuilder.addService(service)
+
+      val damlDef = service.bindService()
+      val damlDesc = damlDef.getServiceDescriptor
+      // Only add "proxy" services if it actually contains com.daml in the service name.
+      // There are other services registered like the reflection service, that doesn't need the special treatment.
+      if (damlDesc.getName.contains("com.daml")) {
+        val digitalassetName = damlDesc.getName.replace("com.daml", "com.digitalasset")
+        val digitalassetDef = ServerServiceDefinition.builder(digitalassetName)
+        damlDef.getMethods.forEach { methodDef =>
+          val damlMethodDesc = methodDef.getMethodDescriptor
+          val digitalassetMethodName =
+            damlMethodDesc.getFullMethodName.replace("com.daml", "com.digitalasset")
```

Comment:
I think a regex might help you reduce duplication.

I haven't tested the below code; apply it at your own peril.

```suggestion
      val match = "^com\\.daml\\.".r
      val replacement = "com.digitalasset."
      if (match.matches(damlDesc.getName)) {
        val digitalassetName = match.replaceFirstIn(damlDesc.getName, replacement)
        val digitalassetDef = ServerServiceDefinition.builder(digitalassetName)
        damlDef.getMethods.forEach { methodDef =>
          val damlMethodDesc = methodDef.getMethodDescriptor
          val digitalassetMethodName =
            match.replaceFirstIn(damlMethodDesc.getFullMethodName, replacement)
```

Reply:
I can pull out the strings into variables, but not sure using regex is improving it.

Reply:
Just makes sure it's at the start of the string, but I'm easy either way.